src/backends/beamcoder.ts

```diff
@@ -399,8 +402,20 @@ export class BeamcoderExtractor extends BaseExtractor implements Extractor {
             this.#packetReadCount++;
         }
 
+        // we read through all the available packets and frames, but we still don't have a frame. This can happen
+        // when our targetPTS is to close to the end of the video. In this case, we'll try to seek further away from
+        // the end of the video and try again. We've set up a MAX_RECURSION to prevent an infinite loop.
         if (!outputFrame) {
-            throw Error('No matching frame found');
+            if (MAX_RECURSION < this.#recursiveReadCount) {
+                throw Error('No matching frame found');
+            }
+            const TIME_OFFSET = 0.1; // time offset in seconds
```

Comment:
Would `0.1` work well for a variety of files?

I wonder if we could just seek here instead of having an offset.

Reply:
Overall, it seems like an approach that works for some files, but not all files. Maybe we can also merge like this and refactor as we gather more edge cases?

Reply:
> Would 0.1 work well for a variety of files? 

I'm not sure if `0.1` will be enough. It depends on how the video is encoded. TBH, the video I added to our tests is pretty weird. We'll have to see. The downside of using a larger offset is that we'll have to read more packets. If we see this issue again, the first thing we should do is either increase this number or the number of recursions.

> I wonder if we could just seek here instead of having an offset.

Since we're at the end of our process, we need to restart the entire process again.